experimental/unitsconversion/src/converter.rs

```diff
@@ -158,4 +177,176 @@ impl<'data> ConverterFactory<'data> {
             Convertibility::NotConvertible
         }
     }
+
+    fn apply_si_prefix(si_prefix: &SiPrefix, ratio: &mut Ratio<BigInt>) {
+        match si_prefix.base {
+            Base::Decimal => {
+                *ratio *= Ratio::<BigInt>::from_integer(10.into()).pow(si_prefix.power as i32);
+            }
+            Base::Binary => {
+                *ratio *= Ratio::<BigInt>::from_integer(2.into()).pow(si_prefix.power as i32);
+            }
+        }
+    }
+
+    fn apply_power(power: i8, ratio: &mut Ratio<BigInt>) {
+        *ratio = ratio.pow(power as i32);
+    }
+
+    fn add_term(
+        &self,
+        unit_item: &MeasureUnitItem,
+        sign: i8,
+        conversion_rate: &mut Ratio<BigInt>,
+    ) -> Result<(), ConversionError> {
+        let conversion_info = self
+            .payload
+            .convert_infos
+            .get(unit_item.unit_id as usize)
+            .ok_or(ConversionError::DataNotFoundError)?;
+
+        let mut conversion_info_factor = Self::extract_ratio_from_unaligned(
+            &conversion_info.factor_sign,
+            conversion_info.factor_num(),
+            conversion_info.factor_den(),
+        );
+
+        Self::apply_si_prefix(&unit_item.si_prefix, &mut conversion_info_factor);
+        Self::apply_power(unit_item.power * sign, &mut conversion_info_factor);
+
+        *conversion_rate *= conversion_info_factor;
+
+        Ok(())
+    }
+
+    fn extract_ratio_from_unaligned(
+        sign_ule: &SignULE,
+        num_ule: &ZeroSlice<u8>,
+        den_ule: &ZeroSlice<u8>,
+    ) -> Ratio<BigInt> {
+        let sign = match Sign::from_unaligned(*sign_ule) {
+            Sign::Positive => num_bigint::Sign::Plus,
+            Sign::Negative => num_bigint::Sign::Minus,
+        };
```

Comment:
suggestion: you'll likely need this again, consider adding `Sign::as_num_bigint`

Reply:
what do you mean?